synapse/handlers/federation_event.py

```diff
@@ -955,6 +960,20 @@ async def _compute_event_context_with_maybe_missing_prevs(
         seen = await self._store.have_events_in_timeline(prevs)
         missing_prevs = prevs - seen
 
+        # If we've already recently attempted to pull this missing event recently, don't
+        # try it again so soon. Since we have to fetch all of the prev_events, we can
+        # bail early here if we find any to ignore.
+        prevs_to_ignore = await self._store.filter_event_ids_with_pull_attempt_backoff(
+            room_id, missing_prevs
+        )
+        if len(prevs_to_ignore) > 0:
+            raise FederationError(
+                "ERROR",
+                429,
+                "Not attempting to pull event that we already tried to pull recently (backing off).",
+                affected=prevs_to_ignore[0],
+            )
```

Comment:
Instead of re-using `FederationError` here, it seems like it would be wise to introduce a new/custom exception type like `PullAttemptBackoffError`

Reply:
Added `FederationPullAttemptBackoffError`